utils/rpm.mk

```diff
@@ -0,0 +1,103 @@
+# Common definitions for make an RPM, used by both the Agent and the Server.
+#
+# Making a pbench component RPM requires a few steps:
+# 1. Get version number.
+# 2. Update the RPM spec file with that version number etc.
+# 3. Update any GIT submodules.
+# 4. Do a "make install" to a temp directory.
+# 5. Generate a tar ball from the directory.
+# 6. Generate a local SRPM that will be uploaded to COPR for building.
+# 7. Optionally generate a local RPM.
+# 8. Clean up the temp directory
+
+PBENCHTOP := $(shell git rev-parse --show-toplevel)
+TMPDIR = /tmp/opt
+
+prog = pbench-${component}
+VERSION := $(file < ${PBENCHTOP}/${component}/VERSION)
+TBDIR = ${TMPDIR}/${prog}-${VERSION}
+
+RPMDIRS = BUILD BUILDROOT SPECS SOURCES SRPMS RPMS
+
+RPMSRC = ${HOME}/rpmbuild/SOURCES
+RPMSRPM = ${HOME}/rpmbuild/SRPMS
+RPMSPEC = ${HOME}/rpmbuild/SPECS
+
+sha1 := $(shell git rev-parse --short HEAD)
+seqno := $(shell if [ -e ./seqno ] ;then cat ./seqno ;else echo "1" ;fi)
+
+# By default we only build a source RPM
+all: srpm
+
+.PHONY: rpm
+rpm: spec srpm
+	rpmbuild -bb ${RPMSPEC}/${prog}.spec
+
+.PHONY: srpm
+srpm: spec patches tarball
+	rm -f ${RPMSRPM}/$(prog)-*.src.rpm
+	rpmbuild -bs ${RPMSPEC}/${prog}.spec
+
+.PHONY: spec
+spec: rpm-dirs ${prog}.spec.j2
+	if [ -e ./seqno ] ;then expr ${seqno} + 1 > ./seqno ;fi
+	jinja2 ${prog}.spec.j2 -D version=${VERSION} -D gdist=g${sha1} -D seqno=${seqno} > ${RPMSPEC}/${prog}.spec
+	rpmlint ${RPMSPEC}/${prog}.spec
+
+.PHONY: patches
+patches: rpm-dirs
+	if [ -d ./patches ] ;then cp ./patches/* ${RPMSRC} ;fi
+
+.PHONY: tarball
+tarball: rpm-dirs submodules ${subcomps}
+	echo "${sha1}" > ${TBDIR}/${component}/SHA1
+	echo "${seqno}" > ${TBDIR}/${component}/SEQNO
+	tar zcf ${RPMSRC}/${prog}-${VERSION}.tar.gz -C ${TMPDIR} ${prog}-${VERSION}
+	rm -rf ${TMPDIR}
+
+.PHONY: rpm-dirs
+rpm-dirs:
+	mkdir -p $(addprefix ${HOME}/rpmbuild/,${RPMDIRS})
+
+.PHONY: submodules
+submodules:
+	git submodule update --init --recursive
+
+# FIXME:  Do we still need the sed command to correct for Python3?
+#	  (Previous, only the server was doing it, and only for two files.)
```

Comment:
Good question. I remember seeing this and commenting before; I don't remember the answer. The idea of editing files in the RPM builder always bugged me...

Reply:
Turns out that we can use RPM to make that happen with the correct Python RPM building support.

Reply:
Apparently the RPM-building tool itself does it, and this is supposed to avoid conflicts with that....

In any case, Nick is going to try to figure out whether we can dispense with this, now.

Reply:
COPR offered no objections, so I'm going to remove this.